### DISCUSSION TITLE
Unrar has been removed, but I need unrar. Alternatives?

### DISCUSSION THREAD

**Discussion by cryo-kronos** [Homebrew/discussions#285] [Everyday usage] [answered]
So, unrar is gone from Homebrew, apparently because of licensing issues.

However, I need to install unrar on macOS Mojave. What are my alternatives?

**Comment by carlocab** [comment, marked as answer]
I've extracted the formula to my tap, which you can use via:
```
brew install carlocab/personal/unrar
```
There are bottles for Catalina and Big Sur, but I guess that doesn't quite work for you, so I suspect it'll build from source. If it doesn't work I'm willing to accept PRs, but I can't promise I'll spend time investigating it myself to fix it.

Otherwise, you can also extract the formula to your own tap yourself using `brew extract`.

**Reply by carlocab** [reply]
Also, there are about a dozen other formulae on homebrew-core that also extract various forms of compressed files. Depending on the formats you are working with, you may find that there is already an existing homebrew-core formula that suits your needs.

**Reply by cryo-kronos** [reply]
> I've extracted the formula to my tap, which you can use via:
> 
> ```
> brew install carlocab/personal/unrar
> ```
> 
> There are bottles for Catalina and Big Sur, but I guess that doesn't quite work for you, so I suspect it'll build from source. If it doesn't work I'm willing to accept PRs, but I can't promise I'll spend time investigating it myself to fix it.
> 
> Otherwise, you can also extract the formula to your own tap yourself using `brew extract`.

Thank you. That works for now.

**Reply by vom513** [reply]
Just came here after googling this.  As was mentioned, other utils will handle rar (obviously more exotic archive operations may vary...).

I just tried 7z (formula: p7zip) and it supports rar archives.

**Reply by yuconora** [reply]
there is another in homebrew-cask
[Named rar](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/rar.rb)
so, you can use
```ruby
brew install rar
```
command `rar` and `unrar` will be found in /usr/local/bin/

**Reply by ShanJiang21** [reply]
> brew install carlocab/personal/unrar

Figured out why there is always a bug saying like below:

> Error: No similarly named formulae found.
> Error: No available formula or cask with the name "unrar".

License sucks.

**Reply by daleyzou** [reply]
it really works, thank you

**Reply by TheShubham-K** [reply]
@cryo-kronos thank you, it really works.

**Reply by thorinmesser** [reply]
> there is another in homebrew-cask [Named rar](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/rar.rb) so, you can use
> 
> ```ruby
> brew install rar
> ```
> 
> command `rar` and `unrar` will be found in /usr/local/bin/

Anyone else coming here from the future will find that rar is now also gone.

**Reply by codymoorhouse** [reply]
> Anyone else coming here from the future will find that rar is now also gone.

Perhaps 😂 


I'll share what I used as a new replacement, it's `unrar` without an `r` but works the same.

```shell
brew install unar
unar file.r00
```

**Reply by gromgit** [reply]
> Anyone else coming here from the future will find that rar is now also gone.

```bash
% brew info --cask rar
==> rar: 7.01
https://www.rarlab.com/
Not installed
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/r/rar.rb
==> Name
RAR Archiver
==> Description
Archive manager for data compression and backups
==> Artifacts
rar/default.sfx -> /opt/homebrew/lib/default.sfx (Generic Artifact)
rar/rarfiles.lst -> /opt/homebrew/etc/rarfiles.lst (Generic Artifact)
rar/rar (Binary)
rar/unrar (Binary)
==> Analytics
install: 5,861 (30 days), 17,349 (90 days), 64,886 (365 days)
```
Seems alive and well to me.

**Reply by udance4ever** [reply]
> Seems alive and well to me.

not so well in May 2025 on two of our Macs, `brew` now installs v7.11 of `rar` and the binaries are just killed (even if I grant `rar` permission in System Settings):
```
% brew upgrade rar
==> Upgrading 1 outdated package:
rar 7.01 -> 7.11
==> Upgrading rar
==> Downloading https://www.rarlab.com/rar/rarmacos-arm-711.tar.gz
######################################################################### 100.0%
==> Backing Generic Artifact 'default.sfx' up to '/opt/homebrew/Caskroom/rar/7.0
==> Removing Generic Artifact '/opt/homebrew/lib/default.sfx'
==> Backing Generic Artifact 'rarfiles.lst' up to '/opt/homebrew/Caskroom/rar/7.
==> Removing Generic Artifact '/opt/homebrew/etc/rarfiles.lst'
==> Unlinking Binary '/opt/homebrew/bin/rar'
==> Unlinking Binary '/opt/homebrew/bin/unrar'
==> Moving Generic Artifact 'default.sfx' to '/opt/homebrew/lib/default.sfx'
==> Moving Generic Artifact 'rarfiles.lst' to '/opt/homebrew/etc/rarfiles.lst'
==> Linking Binary 'rar' to '/opt/homebrew/bin/rar'
==> Linking Binary 'unrar' to '/opt/homebrew/bin/unrar'
==> Purging files for version 7.01 of Cask rar
🍺  rar was successfully upgraded!
% unrar
zsh: killed     unrar
% rar
zsh: killed     rar
```
good news is the workaround to use [Carlo's personal tap](https://github.com/orgs/Homebrew/discussions/285#discussioncomment-206369) works great on both a 2020 M1 MacBookPro (macOS 15.4.1) and an 2017 iMac (macOS 13.7.5) - thanks!

**Reply by bric3** [reply]
FYI `rar` has been disabled as well because it doesn't pass the Gatekeeper checks. 

Duckduckgo brought up https://github.com/ustinka/homebrew-rar

**Comment by cryo-kronos** [comment]
Interestingly, MacPorts still has unrar; they haven’t removed it. Given that MacPorts has traditionally been quite conservative, I’m surprised that Homebrew has removed unrar but MacPorts hasn’t. 

As for alternatives, certain apps (e.g., nzbget) require unrar to function, so 7z isn’t going to work.

**Reply by gromgit** [reply]
> Interestingly, MacPorts still has unrar; they haven’t removed it. Given that MacPorts has traditionally been quite conservative, I’m surprised that Homebrew has removed unrar but MacPorts hasn’t.

MacPorts allows distribution of software under a [*Restrictive/Distributable* license](https://trac.macports.org/wiki/PortfileRecipes#license); Homebrew [doesn't](https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae). Each party is being perfectly consistent in applying their respective rules.

> certain apps (e.g., nzbget) require unrar to function

@carlocab already offered an alternative that neither the Homebrew software nor maintainers have any issue with.

**Reply by reagle** [reply]
Has anyone found any alternatives yet? 7zip says unsupported format.

**Reply by SMillerDev** [reply]
There are a bunch of alternatives in the accepted answer.

**Reply by reagle** [reply]
Sorry, @yuconora's answer was hidden beneath a "View more" which surprised me -- Don't think I've seen that before on Github. Perhaps it's a Discussions thing?

<img width="795" alt="Screen Shot 2021-01-23 at 12 04 04 PM" src="https://user-images.githubusercontent.com/596814/105608800-3531a400-5d73-11eb-84fb-f7b79a7a65e0.png">

**Comment by csarmiento** [comment]
I just found this gem:

```
brew info rar
rar: 6.0.0
https://www.rarlab.com/
Not installed
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/rar.rb
==> Name
RAR Archiver
==> Description
Archive manager for data compression and backups
==> Artifacts
rar/default.sfx -> /usr/local/lib/default.sfx (Generic Artifact)
rar/rarfiles.lst -> /usr/local/etc/rarfiles.lst (Generic Artifact)
rar/rar (Binary)
rar/unrar (Binary)
==> Analytics
install: 2,326 (30 days), 4,512 (90 days), 8,555 (365 days)
```

It seems that rarlab has provided an official `rar/unrar` commad for MacOS

**Reply by bernadinm** [reply]
Nice! Thanks for sharing.

#### Install 

```sh
brew install rar
````

#### Quick Tip

```sh
unrar x "/your/file/here"
```
Enjoy! :)

**Reply by setanarut** [reply]
Do you know open source rar extractor for macOS? this is trial version

```shell
➜  Desktop rar

RAR 6.00   Copyright (c) 1993-2020 Alexander Roshal   1 Dec 2020
Trial version             Type 'rar -?' for help

Usage:     rar <command> -<switch 1> -<switch N> <archive> <files...>
               <@listfiles...> <path_to_extract\>
```

**Reply by setanarut** [reply]
thanks but the freeware binary is not safe. 
I found this.
`brew install unar`
https://github.com/ashang/unar#build-on-macos

**Reply by setanarut** [reply]
Also `brew install p7zip` but license is conflicting.

<img width="505" alt="Ekran Resmi 2021-02-13 06 21 25" src="https://user-images.githubusercontent.com/36481442/107840440-bce35f00-6dc3-11eb-8809-f1dd0b5ce9f2.png">

<img width="604" alt="a" src="https://user-images.githubusercontent.com/36481442/107840424-a50bdb00-6dc3-11eb-8a82-79f76d9a589b.png">

**Reply by ahuigo** [reply]
> I just found this gem:
> 
> ```
> brew info rar
> rar: 6.0.0
> https://www.rarlab.com/
> Not installed
> From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/rar.rb
> ==> Name
> RAR Archiver
> ==> Description
> Archive manager for data compression and backups
> ==> Artifacts
> rar/default.sfx -> /usr/local/lib/default.sfx (Generic Artifact)
> rar/rarfiles.lst -> /usr/local/etc/rarfiles.lst (Generic Artifact)
> rar/rar (Binary)
> rar/unrar (Binary)
> ==> Analytics
> install: 2,326 (30 days), 4,512 (90 days), 8,555 (365 days)
> ```
> 
> It seems that rarlab has provided an official `rar/unrar` commad for MacOS

“unrar” cannot be opened because the developer cannot be verified.

**Reply by csarmiento** [reply]
> Do you know open source rar extractor for macOS? this is trial version
> 
> ```shell
> ➜  Desktop rar
> 
> RAR 6.00   Copyright (c) 1993-2020 Alexander Roshal   1 Dec 2020
> Trial version             Type 'rar -?' for help
> 
> Usage:     rar <command> -<switch 1> -<switch N> <archive> <files...>
>                <@listfiles...> <path_to_extract\>
> ```

However, the `unrar` execuable says:

```
▶ unrar

UNRAR 6.00 freeware      Copyright (c) 1993-2020 Alexander Roshal

Usage:     unrar <command> -<switch 1> -<switch N> <archive> <files...>
               <@listfiles...> <path_to_extract\>

```

**Comment by setanarut** [comment]
```
brew install unar
```

## All alternatives 

- `brew install rar` ( Trial version | Copyright (c) 1993-2020 Alexander Roshal )
- `brew install p7zip` ( Only extraction | Copyright (c) 1999-2016 Igor Pavlov )
- `brew install unar`( Only extraction |  License: LGPL-2.1-or-later )

**Comment by srigi** [comment]
All the proposed alternatives cannot extract RAR5 archives. I downloaded `unrar` binary from https://www.rarlab.com/rar_add.htm and put it into `/usr/local/bin` manually.

**Reply by csarmiento** [reply]
Did you try the `unrar` that comes with **rar** from brew (`brew install rar`) ? 

As the command says it is freeware...

```
▶ unrar

UNRAR 6.01 freeware      Copyright (c) 1993-2021 Alexander Roshal

Usage:     unrar <command> -<switch 1> -<switch N> <archive> <files...>
               <@listfiles...> <path_to_extract\>
```

**Reply by srigi** [reply]
I didn't. My problem is now solved, so nothing more I need to do.
Just for the sake of persistence (like when I reinstall macOS), I've put binary into my custom bin `$HOME/bin`.

But thank you for the suggestion.

**Comment by mrochcn** [comment]
brew install rar

**Comment by nguyenvulong** [comment]
`7z` did not work for me, but `The Unarchiver` (`brew install --cask the-unarchiver`) did.

**Comment by Bo98** [comment]
Worth noting that system `tar` on macOS does have limited support for extracting rar files. It may be enough for some use cases:

```
tar xf sample.rar
```

It doesn't support anything using filters or encryption.

macOS's `tar` is an older version of `libarchive`'s `bsdtar`. You can `brew install libarchive` for a newer version (named `bsdtar` instead) with better RARv5 and filter support which we've been using for `rar` files for a while internally with decent success: https://github.com/Homebrew/brew/blob/673f19086b0909b13664b2a3873f26025ad5236c/Library/Homebrew/unpack_strategy/rar.rb#L28-L31. Apple's version will probably eventually gain this support when they update to something newer than their current 3.5.3.

If it doesn't work for the type of rar you're using then one of the other solutions in this thread will work: `unar` formula, `7zip` formula or some casks like `rar` and `the-unarchiver`.

For creating rars, you have no options except the `rar` cask due to strict legal protections on who can create rar files.

**Reply by Bo98** [reply]
As an update here: macOS 15.4 ships an updated `tar` so installing `libarchive` shouldn't be necessary anymore for many basic extraction cases if you use the latest macOS.

**Reply by ben-xo** [reply]
@Bo98 Thanks for this - excellent intel

**Comment by bitplane** [comment]
I've been working on this if anyone's interested. It's a free software rar alternative that supports compression:

https://github.com/bitplane/rars
